.github/dependabot.yaml

```diff
@@ -0,0 +1,29 @@
+version: 2
+updates:
+  # GitHub Actions
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "weekly"
+      day: "monday"
+    commit-message:
+      prefix: "chore:"
+
+  - package-ecosystem: "gomod"
+    directory: "/"
+    schedule:
+      interval: "weekly"
+      day: "tuesday"
+    groups:
+      kubernetes:
+        patterns: [ "k8s.io/*" ]
+    ignore:
+      # Ignore k8s-operator-libs, controller-runtime, and Kubernetes major and minor updates. These should be done manually.
+      - dependency-name: "sigs.k8s.io/controller-runtime"
+        update-types: [ "version-update:semver-major", "version-update:semver-minor" ]
+      - dependency-name: "github.com/NVIDIA/k8s-operator-libs"
+        update-types: [ "version-update:semver-major", "version-update:semver-minor" ]
+      - dependency-name: "k8s.io/*"
+        update-types: [ "version-update:semver-major", "version-update:semver-minor" ]
```

Comment:
These ignores mean we will have to manually update (for major and minor releases) K8s-operator-libs, controller runtime and all K8s dependencies. I think this makes sense as this is a huge part of our dependency tree and we might prefer to take it on manually.

Controller runtime also regularly enough requires some kind of code changes for a minor release so the likelihood is we'd have to do this anyway. And controller runtime has one release per K8s release.